Assets/_BForBoss/_Core/Scripts/Editor/LevelDesignFeedback/LevelDesignFeedbackEditorWindow.cs

```diff
@@ -0,0 +1,213 @@
+                
+                if (_image == null)
+                {
+                    EditorGUI.DrawRect(previewImageRect, Color.gray);
+                }
+                else
+                {
+                    EditorGUI.DrawTextureTransparent(previewImageRect, _image, ScaleMode.ScaleToFit, 0);
+                }
+
+                if (_image != null && WasElementDoubleClicked(previewImageRect))
+                {
+                    // new SceneViewCameraWindow(SceneView.currentDrawingSceneView)
+
+                    ImageEditorWindow imagePopupWindow = GetWindow<ImageEditorWindow>();
+                    imagePopupWindow.OnWindowClosed += OnImageEdited;
+                    imagePopupWindow.OpenWindow(_image);
+                }
+            
+                GUILayout.FlexibleSpace();
+            }
+            
+            EditorGUILayout.Space(previewImageRect.y + previewImageRect.height);
+            
+            using (new EditorGUILayout.HorizontalScope())
+            {
+                GUILayout.FlexibleSpace();
+
+                if (GUILayout.Button("Reset"))
+                {
+                    _image = CreateTextureCopy(_originalImage);
+                }
+                
+                EditorGUILayout.Space(position.width * 0.05f);
+            }
+        }
+        
+        EditorGUILayout.Space(ELEMENT_SPACING);
+    }
+
+    private void OnImageEdited(Texture2D editedScreenShot)
+    {
+        _image = editedScreenShot;
+        Repaint();
+    }
+
+    private void DrawDescription()
+    {
+        using (new EditorGUILayout.HorizontalScope())
+        {
+            float labelWidth = EditorGUIUtility.labelWidth;
+            EditorGUIUtility.labelWidth = 35f;
+            EditorGUILayout.PrefixLabel("Title", GUI.skin.button, EditorStyles.boldLabel);
+            _title = EditorGUILayout.TextField(_title, GUILayout.ExpandWidth(true));
+            EditorGUIUtility.labelWidth = labelWidth;
+        }
+            
+        EditorGUILayout.Space(ELEMENT_SPACING);
+        EditorGUILayout.LabelField("Feedback",EditorStyles.boldLabel);
+        _feedback = EditorGUILayout.TextArea(_feedback, GUI.skin.textArea, GUILayout.Height(EditorGUIUtility.singleLineHeight * NUMBER_OF_FEEDBACK_LINES));
+    }
+
+    private void DrawButtons()
+    {
+        using (new EditorGUILayout.HorizontalScope())
+        {
+            EditorGUILayout.Space(ELEMENT_SPACING);
+
+            if (GUILayout.Button("Close"))
+            {
+                _window.Close();
+            }
+                
+            GUILayout.FlexibleSpace();
+
+            GUI.enabled = _image != null && !string.IsNullOrEmpty(_title) && !string.IsNullOrEmpty(_feedback);
+                
+            if (GUILayout.Button("Create Feedback"))
+            {
+                SendFeedback();
+            }
+
+            GUI.enabled = true;
+
+            EditorGUILayout.Space(ELEMENT_SPACING);
+        }
+    }
+
+    private void SendFeedback()
```

Comment:
https://github.com/PerigonGames/BForBoss/pull/141/files#r828761896

Basically, everything from here downwards, can be abstracted out to its own separate logic class.